### PR TITLE
Re-add raw methods, accept leveldb in constructor, minor fixes

### PR DESCRIPTION
**Pull request by s1na**
**tl;dr** Changes since the `v3.0.0` release had introduced breaking changes (and bugs). This PR makes those changes backwards compatible, and fixes the bugs. `ethereumjs-vm` passes all tests without any modifications after integrating this PR.

The changes are:

- Re-introduce `getRaw`, `putRaw`, `delRaw` methods (but mark them as deprecated)
- Instead of a `DB` object, instantiate the trie via a leveldb object (like before). This necessitated some other changes in `checkpointTrie` and `secureTrie`.
- Fixed a bug in `CheckpointTrie._exitCpMode` which passed the wrong object to `WriteStream`
- The first two changes created a (difficult to track) bug: `SecureTrie.copy` would copy `CheckpointTrie`'s scratch db and use that as the underlying database, which meant the actual state data wasn't available and this caused quite a few failures.

### CODE REVIEW

File: src/baseTrie.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ module.exports = class Trie {
     this.EMPTY_TRIE_ROOT = ethUtil.KECCAK256_RLP
     this.sem = semaphore(1)
 
-    this.db = db || new DB()
+    this.db = db ? new DB(db) : new DB()
```
Comment (holgerd77) [inline src/baseTrie.js]: Associated JSDocs need to be updated as well.

File: src/baseTrie.js [hunk 2]
```diff
@@ -162,6 +162,31 @@ module.exports = class Trie {
...
+  }
+
+  /**
+   * Deletes key directly from underlying key/value db.
+   * @deprecated
+   */
+  delRaw (key, cb) {
+    this.db.del(key, cb)
+  }
+
```
Comment (holgerd77) [inline src/baseTrie.js]: Ok.

File: src/baseTrie.js (hunk 1 above)
Comment (s1na) [reply src/baseTrie.js]: Good catch. Also had forgotten to re-generate the docs.

**Comment by holgerd77** [comment]
@s1na Phew, I am just realizing while scimming through the [merged PR history](https://github.com/ethereumjs/merkle-patricia-tree/pulls?q=is%3Apr+is%3Aclosed) that doing releases on this library got completely lost. 😕 Last release (`v3.0.0`) was from January 2019.

Do you think it is still worth to do a pre-TypeScript `v3.1.0` release or should we rather get the refactoring changes @ryanio is doing right now together and do a bigger `v4.0.0` release? Any opinion or preference? Could we in that case branch off for such a release from this PR?

**Comment by s1na** [comment]
@holgerd77 that's not a bad idea actually. It seems this PR is backwards-compatible (this should be tested again though).

**Comment by ryanio** [comment]
I realized I removed the `@deprecated` `getRaw/putRaw/delRaw` methods during the refactor in #107, so just a note that they won't be included in `v4.0.0`. If there is a good reason for them to be included please let me know :)
